### ISSUE TITLE
Invalid Params Crash

### ISSUE THREAD

**Issue by PhilCK** [intel/opencl-intercept-layer#89]
## Observed Behavior
In our Unit testing we have a lot of tests that try to weed out bad parameters. When running through the intercept layer these bad params can take down the application from the intercept layer.

clCreateProgramWithSource with bad params crash on these lines.

```
// intercept.cpp
size_t length = 0;
if( ( lengths == NULL ) ||
  ( lengths[i] == 0 ) )
{  
  length = strlen( strings[i] );
}
```

clCreateSubBuffer with junk crashes on pRegion->origin

```
// intercept.cpp
{
  cl_buffer_region* pRegion = (cl_buffer_region*)createInfo;
  ss << "origin = "
  << pRegion->origin
  << " size = "
  << pRegion->size;
}
```

clGetExtensionFunctionAddress* with nullptr gets passed to func_name;

```
void* CLIntercept::getExtensionFunctionAddress(cl_platform_id platform, const std::string& func_name);
```
## Desired Behavior

Not crashing.

## Steps to Reproduce

- Passing junk values to the above entry points.
- Internal OCL

**Comment by PhilCK** [comment]
I think we are happy to add the checks if they are desired.

**Comment by bashbaug** [comment]
Thanks!  I think I've fixed all of these.  If you find any more, or if there's some aspect of the cases described above that I haven't handled, feel free to file another issue (or re-open this one).

> I think we are happy to add the checks if they are desired.

This would be great too - I'm always happy to take pull requests.  Hopefully I've managed to save you a bit of time though.

Thanks again for the report!
